riscv/triggers.cc

```diff
@@ -17,6 +17,51 @@ action_t trigger_t::legalize_action(reg_t val) const noexcept {
   return (val > ACTION_MAXVAL || (val == ACTION_DEBUG_MODE && get_dmode() == 0)) ? ACTION_DEBUG_EXCEPTION : (action_t)val;
 }
 
+unsigned trigger_t::legalize_mhselect(reg_t val, bool h_enabled) const noexcept {
+  switch (val) {
+    case MHSELECT_IGNORE:
+    case MHSELECT_MCONTEXT:
+      return val;
+    case 1:
+    case 2:
+      if (!h_enabled)
+        return MHSELECT_IGNORE;
+      return val;
+    case 3: return MHSELECT_IGNORE;
+    case 5:
+    case 6:
+      if (!h_enabled)
+        return MHSELECT_MCONTEXT;
+      return val;
+    case 7: return MHSELECT_IGNORE;
+    default: assert(false);
+  }
```

Comment:
This is still messy and still embeds assumptions about the values of the enums. I was thinking something more like:
```
  unsigned convert[8] = {
    MHSELECT_IGNORE,  // 0                                                                                                                          
    h_enabled ? MHSELECT_MCONTEXT_SELECT : MHSELECT_IGNORE,  // 1                                                                                   
    h_enabled ? MHSELECT_VMID_SELECT : MHSELECT_IGNORE,  // 2                                                                                       
    MHSELECT_IGNORE,  // 3                                                                                                                          
    MHSELECT_MCONTEXT,  // 4                                                                                                                        
    h_enabled ? 5 : MHSELECT_MCONTEXT,  // 5                                                                                                        
    h_enabled ? 6 : MHSELECT_MCONTEXT,  // 6                                                                                                        
    MHSELECT_IGNORE  // 7                                                                                                                           
  };
  assert(val < 8);
  return convert[val];
```

But I still don't like having 5 & 6 in there instead of always returning an enum value.

Reply:
Is the following better?
```
  mask = h_enabled ? (MHSELECT_MCONTEXT | MHSELECT_MCONTEXT_SELECT | MHSELECT_VMID_SELECT) : (MHSELECT_MCONTEXT);
  val = val & mask;
  return val == MHSELECT_RESERVED ? MHSELECT_IGNORE : val;
```

Reply:
Is the following better?
```
  mask = h_enabled ? (MHSELECT_MCONTEXT | MHSELECT_MCONTEXT_SELECT | MHSELECT_VMID_SELECT) : (MHSELECT_MCONTEXT);
  val = val & mask;
  return val == MHSELECT_RESERVED ? MHSELECT_IGNORE : val;
```